src/test/java/uk/gov/hmcts/reform/civil/service/PaymentsServiceTest.java

```diff
@@ -152,4 +175,99 @@ private CreditAccountPaymentRequest getExpectedCreditAccountPaymentRequest(CaseD
+            );
+            assertThat(exception.getMessage()).isEqualTo(FEE_NOT_SET_CORRECTLY_ERROR);
+        }
+
+        @Test
+        void validateRequestShouldThrowAnError_whenFeeDetailsDoNotHaveFeeCode() {
+            CaseData caseData = CaseData.builder()
+                .hearingFeePBADetails(HFPbaDetails.builder()
+                                                     .fee(Fee.builder()
+                                                              .calculatedAmountInPence(BigDecimal.valueOf(10800))
+                                                              .version("1")
+                                                              .build())
+                                                     .build())
+                .build();
+
+            Exception exception = assertThrows(
+                InvalidPaymentRequestException.class,
+                () -> paymentsService.validateRequest(caseData)
+            );
+            assertThat(exception.getMessage()).isEqualTo(FEE_NOT_SET_CORRECTLY_ERROR);
+        }
+
+        @Test
+        void validateRequestShouldThrowAnError_whenFeeDetailsDoNotHaveFeeVersion() {
+            CaseData caseData = CaseData.builder()
+                .hearingFeePBADetails(HFPbaDetails.builder()
+                                                     .fee(Fee.builder()
+                                                              .calculatedAmountInPence(BigDecimal.valueOf(10800))
+                                                              .code("FEE0442")
+                                                              .build())
+                                                     .build())
+                .build();
+
+            Exception exception = assertThrows(
+                InvalidPaymentRequestException.class,
+                () -> paymentsService.validateRequest(caseData)
+            );
+            assertThat(exception.getMessage()).isEqualTo(FEE_NOT_SET_CORRECTLY_ERROR);
+        }
+
+        @Test
+        void validateRequestShouldThrowAnError_whenFeeDetailsDoNotHaveFeeAmount() {
+            CaseData caseData = CaseData.builder()
+                .hearingFeePBADetails(HFPbaDetails.builder()
+                                                     .fee(Fee.builder()
+                                                              .code("FEE0442")
+                                                              .version("1")
+                                                              .build())
+                                                     .build())
+                .build();
+
+            Exception exception = assertThrows(
+                InvalidPaymentRequestException.class,
+                () -> paymentsService.validateRequest(caseData)
+            );
+            assertThat(exception.getMessage()).isEqualTo(FEE_NOT_SET_CORRECTLY_ERROR);
+        }
+
+        @Test
+        void shouldCreatePaymentServiceRequest_whenValidCaseDetails() {
```

Comment:
This unit test failed with a NullPointerException to me. Please take a look.

Here's a portion of the stack trace for your convenience:
```
java.lang.NullPointerException
	at uk.gov.hmcts.reform.civil.service.PaymentsService.createCreditAccountPayment(PaymentsService.java:85)
	at uk.gov.hmcts.reform.civil.service.PaymentsServiceTest$Specified.shouldCreateCreditAccountPayment_whenValidCaseDetails(PaymentsServiceTest.java:156)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        [...]
```